scheduler/scheduler.go

```diff
@@ -189,6 +189,21 @@ func (s *Scheduler) Sync(ctx context.Context, client schema.ClientMeta, tables s
 
 	// send migrate messages first
 	for _, table := range tables.FlattenTables() {
+		if syncClient.deterministicCQID {
+			for i, c := range table.Columns {
+				if !c.PrimaryKey {
+					continue
+				}
+				if c.Name == schema.CqIDColumn.Name {
+					// CQ_ID should be PK and not have a Unique Clause on it
+					table.Columns[i].PrimaryKey = true
+					table.Columns[i].Unique = false
```

Comment:
Modifying the result of `FlattenTables` won't make it true for the real tables list so they'll still 'see it' as `Unique: true`. I have no recommendation on how to solve this off the top of my head.

Reply:
But only the table definition in the migration event  actually changes the schema... This enable all of the docs to remain unchanged

Reply:
_but_ in the docs `cq_id` will show up as unique, which would be incorrect (since the source itself always specifies DetCQId:true now...)
Still I don't think this is an issue

Reply:
Just looked at the `plugin-sdk/v4/docs` package and we only show in our docs columns that are incremental or that have a PK so the `Unique` clause is never actually shown to the user unless they go to the schema of one of the 4 dbs that actually supports it